Removed additional output workspaces from LoadNexusMonitors

MikeHart85 opened the pull request:
Fixes #13032.

Removed the code that was adding an output property for each child workspace. Since the OutputWorkspace is a group, all of its children are exported anyway. The return value from the Python interface is now just a workspace group with its children contained within.

This does have the side effect of changing the way the child workspaces are named; they are now prefixed by the name of the group (both in the ADS and in Python):
```python
ws_group (
  ws_group_1,
  ws_group_2,
  ws_group_3,
  ws_group_4
)
```
Rather than just `_1`, `_2`, etc.

Not sure whether this can cause any issues? The unit tests pass alright, but if any user scripts expect specific names this would break them. On the other hand, it seems much more likely that they would iterate or access them by index. Please advise.

### Testing
The file `ExternalData/Testing/Data/UnitTest/LARMOR00003368.nxs` will work to demonstrate this.

Script:
```python
# Create using python return value
file_path = "LARMOR00003368.nxs"
ws_group = LoadNexusMonitors(Filename=file_path)
print "Length of the GroupWorkspace is: %s, we expected 4." % len(ws_group)
print "The individual sub-workspace names are:"
for element in ws_group:
    print "Workspace: %s" % element.name()

print '###################################'

# Create by passing in workspace
out_name = "group_ws_test"
LoadNexusMonitors(Filename=file_path, OutputWorkspace=out_name)
ws_group_2 = mtd[out_name]
print "Length of the GroupWorkspace is: %s, we expected 4."  % len(ws_group_2)
print "The individual sub-workspace names are:"
for element in ws_group_2:
    print "Workspace: %s" % element.name()
```

The ouput should now be:
```
Length of the GroupWorkspace is: 4, we expected 4.
The individual sub-workspace names are:
Workspace: ws_group_1
Workspace: ws_group_2
Workspace: ws_group_3
Workspace: ws_group_4
###################################
Length of the GroupWorkspace is: 4, we expected 4.
The individual sub-workspace names are:
Workspace: group_ws_test_1
Workspace: group_ws_test_2
Workspace: group_ws_test_3
Workspace: group_ws_test_4
```

picatess commented:
As Owen suggested, it might be best to create a 2nd version of the algorithm as the interface has changed. It would be also good to document the output format with a small unit test.

MikeHart85 commented:
As discussed, I've created a version 2 of the LoadNexusMonitors algorithm which fixes this issue. I've also modified the first version to just call the new version as a child algorithm and then reproduce the issue, in order to cut down on code repetition.

Added an rst documentation file for version 2 and updated the rst for version 1.

Added a small unit test to document and verify this fix. It ensures that this issue is fixed in version 2, but that version 1 continues behaving as it did before.

picatess commented:
Tested the two versions of the Algorithm and it seems to work.
